frontend/src/context/CartContext.js

```diff
@@ -0,0 +1,80 @@
+import { useEffect, useReducer } from 'react'
+import constate from 'constate'
+import logger from './Logger'
+import useLocalStorage from '../shared/useLocalStorage'
+
+const initialState = {
+  cart: [],
+  total: 0
+}
+
+const reducer = (state, action) => {
+  switch (action.type) {
+    case 'ADD_PRODUCT':
+      const id = action.payload.product.id
+      let addedProduct = state.cart.find(product => product.id === id)
+
+      if(addedProduct) {
+        addedProduct.quantity += 1
```

Comment:
we will also have a quantity selector in the frontend. How does that work then, do we call add_product 5 times in a row? :D

Reply:
that will require a new action i.e `SET_PRODUCT` or `SET_QUANTITY`